hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -297,7 +305,15 @@ def do_draw(self, data):
             return result
 
     def filter(self, condition):
+        # Handle a few specific weird cases.
+        if condition is math.isfinite:
+            return self
+        if condition in [math.isinf, math.isnan]:
+            return nothing()
```

Comment:
Not correct for floats!  Adjust min and max bounds in the former case; use `self.permitted(...)` and `st.nothing()`/`st.just()`/`st.sampled_from()` as appropriate.